cmd/otel-allocator/config/config.go

```diff
@@ -39,17 +40,17 @@ const DefaultConfigFilePath string = "/conf/targetallocator.yaml"
 const DefaultCRScrapeInterval model.Duration = model.Duration(time.Second * 30)
 
 type Config struct {
-	ListenAddr             string             `yaml:"listen_addr,omitempty"`
-	KubeConfigFilePath     string             `yaml:"kube_config_file_path,omitempty"`
-	ClusterConfig          *rest.Config       `yaml:"-"`
-	RootLogger             logr.Logger        `yaml:"-"`
-	LabelSelector          map[string]string  `yaml:"label_selector,omitempty"`
-	PromConfig             *promconfig.Config `yaml:"config"`
-	AllocationStrategy     *string            `yaml:"allocation_strategy,omitempty"`
-	FilterStrategy         *string            `yaml:"filter_strategy,omitempty"`
-	PrometheusCR           PrometheusCRConfig `yaml:"prometheus_cr,omitempty"`
-	PodMonitorSelector     map[string]string  `yaml:"pod_monitor_selector,omitempty"`
-	ServiceMonitorSelector map[string]string  `yaml:"service_monitor_selector,omitempty"`
+	ListenAddr             string               `yaml:"listen_addr,omitempty"`
+	KubeConfigFilePath     string               `yaml:"kube_config_file_path,omitempty"`
+	ClusterConfig          *rest.Config         `yaml:"-"`
+	RootLogger             logr.Logger          `yaml:"-"`
+	CollectorSelector      metav1.LabelSelector `yaml:"collector_selector,omitempty"`
```

Comment:
should this be reference? or should we remove `omitempty`? I'm not sure the TA can work without the collector selector.

Reply:
It should be, good catch. The semantics of `v1.LabelSelector`, as per the [documentation](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), are: 

> A label selector is a label query over a set of resources. The result of matchLabels and matchExpressions are ANDed. An empty label selector matches all objects. A null label selector matches no objects.

So the target allocator should in fact work without a selector, it'll behave the same way as if there were no collectors to assign targets to.